### PR TITLE
Add list_by_scope

### LINKED ISSUE #5751: [ManagementLockClient] Add list_locks_by_scope API in swagger
The swagger contains `get/put/delete` `/{scope}/providers/Microsoft.Authorization/locks/{lockName}` for all scope, but nothing for `/{scope}/providers/Microsoft.Authorization/locks`.

This makes SDK auto generated has no method like `list_by_scope()`. Because of this, if I wanna list all locks of a scope, I need to know what's the scope, a resource, resource group or a  subscription.

### PR DESCRIPTION
**Pull request by yuwzho**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
Fixes #5751
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5953 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5953/commits/5941a51a476b6b4a2761aa6e716d4156305141c6

**Comment by yuwzho** [comment]
Can anyone help me on the CI? It gives an error message `ApplicationException: Unable to determine SDK generation directory. Exiting....`, I have no idea on it.

**Comment by kpajdzik** [comment]
> Can anyone help me on the CI? It gives an error message `ApplicationException: Unable to determine SDK generation directory. Exiting....`, I have no idea about it.

Don't worry about .NET generation failures

**Review by kpajdzik — APPROVED** [review]
LGTM. It's a simple change but let's wait for ARM team to approve as well.

**Comment by kpajdzik** [comment]
If you haven't taken a look at our new onboarding experience at [OpenAPIHub](http://aka.ms/openapihub), it's a convenient way to create your PR when you're copying from an existing API version or when you're editing your existing specs. If you have any feedback or questions, feel free to use the feedback button on top of the site for help. Thanks!

**Comment by yuwzho** [comment]
@kpajdzik Thank you!

File: specification/resources/resource-manager/Microsoft.Authorization/stable/2016-09-01/locks.json [hunk 1]
```diff
@@ -812,6 +812,46 @@
         },
         "x-ms-odata": "#/definitions/ManagementLockObject"
       }
+    },
+    "/{scope}/providers/Microsoft.Authorization/locks": {
```
Comment (antmarti-microsoft) [inline specification/resources/resource-manager/Microsoft.Authorization/stable/2016-09-01/locks.json]: This swagger already has definitions for list at subscription level, resource group level and extension resource level. Please could you respond to clarify what sort of string you expect {scope} to be?

Comment (yuwzho) [reply specification/resources/resource-manager/Microsoft.Authorization/stable/2016-09-01/locks.json]: For consistency, as all the put/get/delete methods provide subscription, resource group, resources, scope four levels. However, when listing, the scope one is missing.

From user's perspective, all the four levels results in the same URI, I don’t need to care whether the id is resources, resource group or subscription when I use the scope one.

**Review by anthony-c-martin — APPROVED** [review]

**Review by kpajdzik — APPROVED** [review]

**Pull request merged**